Add note about image regeneration in June 2022 (#24524)
mistercrunch/Airflow: docs/docker-stack/changelog.rst

Occasionally our images need to be regenerated using newer ``Dockerfiles`` or constraints.
This happens when an issue is found or a breaking change is released by our dependencies
that invalidates the already released image, and regenerating the image makes it usable again.
While we cannot assure 100% backwards compatibility when it happens, we at least document it
here so that users affected can find the reason for the changes.

+--------------+---------------------+-----------------------------------------+------------------------+----------------------------------------------+
| Date         | Affected images     | Potentially breaking change             | Reason                 | Link to Pull Request                         |
+==============+=====================+=========================================+========================+==============================================+
| 17 June 2022 | 2.2.5               | * The ``Authlib`` library downgraded    | Flask App Builder      | https://github.com/apache/airflow/pull/24516 |
|              |                     |   from 1.0.1 to 0.15.5 version          | not compatible with    |                                              |

> |              |  2.3.0-2.3.2        |                                         | Authlib >= 1.0.0       |                                              |

Reviewer: ```suggestion
|              | 2.3.0-2.3.2         |                                         | Authlib >= 1.0.0       |                                              |
```

Oops, my bad, got a space in the wrong spot.

       any Airflow version from the ``Airflow 2`` line. There is no guarantee that it works, but if it does,
       then you can use latest features from that image to build the previous Airflow versions.

Changes after publishing the images
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Important note!

> 

Reviewer: ```suggestion
```

I think we could get rid of this. I'm not sure it'll really draw any extra attention anyways.

> Some of the images below might be sometimes regenerated using newer ``Dockerfiles`` or constraints.
> This happens when there is an issue found or breaking change released by some of our dependencies
> that invalidates already released image and the images need refreshing. While we cannot assure 100% backwards
> compatibility when it happens, we at least document it here so that users affect might find it out and
> be aware of the changes.

Reviewer: ```suggestion
Occasionally our images need to be regenerated using newer ``Dockerfiles`` or constraints.
This happens when an issue is found or a breaking change is released by our dependencies
that invalidates the already released image, and regenerating the image makes it usable again.
While we cannot assure 100% backwards compatibility when it happens, we at least document it
here so that users affected can find the reason for the changes.
```

nit: I reworded this paragraph, reads better to me.

+--------------+---------------------+-----------------------------------------+------------------------+----------------------------------------------+
| Date         | Affected images     | Potentially breaking change             | Reason                 | Link to Pull Request                         |
+==============+=====================+=========================================+========================+==============================================+
| 17 June 2022 | 2.2.5               | * The ``Authlib`` library downgraded    | Flask App Builder      | https://github.com/apache/airflow/pull/24516 |

> |              | 2.3.0, 2.3.1, 2.3.2 |   from 1.0.1 to 0.15.5 version          | Not compatible with    |                                              |

Reviewer: ```suggestion
|              | 2.3.0-2.3.2         |   from 1.0.1 to 0.15.5 version          | not compatible with    |                                              |
```

Another nit

> |              |                     |                                         | Authlib >= 1.0.0       |                                              |

Reviewer: ```suggestion
|              |                     |   from 1.0.1 to 0.15.5 version          | not compatible with    |                                              |
|              |  2.3.0-2.3.2        |                                         | Authlib >= 1.0.0       |                                              |
```

This should render 2.2.5 and 2.3.0-2.3.2 on separate lines, easier to read (noticed that in the other screenshot you posted).

Author: It will be different rendering in the docs but yeah ...

+--------------+---------------------+-----------------------------------------+------------------------+----------------------------------------------+

> | 18 Jan 2022  | All 2.2.*, 2.1.*    | * The AIRFLOW_GID 500 was removed       | MySQL changed keys     | https://github.com/apache/airflow/pull/20912 |

Reviewer: ```suggestion
| 18 Jan 2022  | All 2.2.\*, 2.1.\*  | * The AIRFLOW_GID 500 was removed       | MySQL changed keys     | https://github.com/apache/airflow/pull/20912 |
```

This should avoid the italics?

Author: Yep: Good one.

<img width="1149" alt="Screenshot 2022-06-23 at 00 35 09" src="https://user-images.githubusercontent.com/595491/175165536-c555325b-5e4c-44a9-bb6f-53f5862c1595.png">
